app/domain/authentication/authn_gcp/validate_resource_restrictions_match_jwt.rb

```diff
@@ -0,0 +1,57 @@
+module Authentication
+  module AuthnGcp
+
+    # This class is responsible of validating resource restrictions values against JWT token.
+    # the assumption is that resource restrictions list contains only permitted types which already validated
+    # in previous steps
+    ValidateResourceRestrictionsMatchJWT = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       %i(resource_restrictions decoded_token restriction_prefix)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnGcp::ValidatingResourceRestrictionsValues.new)
+        validate
+        @logger.debug(LogMessages::Authentication::AuthnGcp::ValidatedResourceRestrictionsValues.new)
+      end
+
+      private
+
+      def validate
+        @resource_restrictions.each do |r|
+          restriction_type = r.type
+          restriction_value = r.value
+          restriction_value_from_token = restriction_value_from_token(restriction_type_without_prefix(restriction_type))
+          unless restriction_value == restriction_value_from_token
+            raise Errors::Authentication::Jwt::InvalidResourceRestrictions.new(restriction_type)
+          end
+          @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictionValue.new(restriction_type))
+        end
+      end
+
+      def restriction_type_without_prefix(restriction_type)
+        restriction_type_without_prefix = restriction_type.sub(@restriction_prefix, "")
+        restriction_type_without_prefix
+      end
```

Comment:
this can be changed to 
```
      def restriction_type_without_prefix(restriction_type)
        restriction_type.sub(@restriction_prefix, "")
      end
```